src/modules/form/hooks/useFormDialog.tsx

```diff
@@ -83,7 +83,12 @@ export function useFormDialog<T extends SubmitFormAllowedTypes>({
       projectId:
         projectId || localConstants?.projectId || inputVariables?.projectId,
     },
-    localDefinition
+    localDefinition,
```

Comment:
I was just fiddling around here, but maybe we should just revert this change. Throwing the error here inside of the hook (as opposed to in a rendered component) just errors in the console. But it will only get thrown visibly to the user when they try to actually render a component that uses the form definition, like on line 130 below.

Reply:
I'm not sure I see the benefit to throwing the error from a callback vs throwing it in the component. Thanks to typescript, I think most of the components that call `useDefinition` already _have to_ deal with the state where `!loading && !formDefinition`--  most of them error in this case, but some render a NotFound component, which I think is wrong (`ViewRecord`, `EditRecord`). In the recent issue that came up-- where a user saw "Not Found" in the case note modal-- it would have been preferable if that had been treated as an error and reported through sentry so we could catch it asap.

I _think_ it would make sense for `useFormDefinition` to just raise an error if the fetch turns up empty. I can't think of any cases where we would expect that, but maybe I'm missing something.

Reply:
Sounds good, I added the throw in `useFormDefinition`. 

I also removed the throw/NotFound returns from the other components, since I thought it's a bit clearer if this situation is only explicitly dealt with in one place. But I wonder if this is a bit messy. I want to think about it more.

Reply:
Hmm I think I see what you mean. I would look at `EditRecord` for example and say "oh, so this component will render null if there is no form" (since it has that `formDefinition && ` conditional render syntax). Which of course isn't true, because that won't get hit. I guess I would almost prefer if BOTH threw an error, so in EditRecord have `if (!formDefinition) throw "form definition not found"` to appease typescript, since at least we're not implying something different with the render. That error wont get hit, of course, because useFormDefinition will have already raised.
I also REALLY like that the query variables are included when raising the error from useFormDefinition, so that will help us track it down. I think that's another "plus" to raising from useFormDefinition.

Reply:
okay, sounds good